scripts/installation/install_solc.py

```diff
@@ -1,69 +1,47 @@
 #!/usr/bin/env python3
 
-
 """
-This file is part of nucypher.
-
-nucypher is free software: you can redistribute it and/or modify
-it under the terms of the GNU Affero General Public License as published by
-the Free Software Foundation, either version 3 of the License, or
-(at your option) any later version.
+Download supported version of solidity compiler for the host operating system.
 
-nucypher is distributed in the hope that it will be useful,
-but WITHOUT ANY WARRANTY; without even the implied warranty of
-MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-GNU Affero General Public License for more details.
-
-You should have received a copy of the GNU Affero General Public License
-along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+This script depends on `nucypher.blockchain.eth.sol` submodule and `py-solc-x` library 
+but otherwise does not require installation of `nucypher` or it's dependencies. 
 """
 
+import os
+from os.path import dirname, abspath
+from pathlib import Path
 
-"""
-Download supported version of solidity compiler for Linux based operating system. 
-This script depends on `nucypher.blockchain.eth.sol` submodule and `requests` library 
-but does not require installation of `nucypher`. 
-"""
-
-import platform
-import shutil
-from os.path import dirname, join, abspath
-from os import chmod
-import requests
+from solcx import install_solc, set_solc_version
 
-PACKAGE_NAME = join('nucypher', 'blockchain', 'eth', 'sol')
-BASE_DIR = dirname(dirname(dirname(abspath(__file__))))
-FILE_PATH = join(BASE_DIR, PACKAGE_NAME, "__conf__.py")
 
-METADATA = dict()
-with open(FILE_PATH) as f:
-    exec(f.read(), METADATA)
+def get_solc_version() -> str:
 
-SOLC_VERSION = METADATA['SOLIDITY_COMPILER_VERSION']
-SOLC_BIN_PATH = join(dirname(shutil.which('python')), 'solc')
+    env_version = os.environ.get('NUCYPHER_SOLIDITY_VERSION')
+    if env_version:
+        return env_version
```

Comment:
Maybe for script use input argument instead of environment variable? Because if somebody wants to just set variable and install specific version instead of recommended easier to use arg

Reply:
:heavy_check_mark: